client/my-sites/design-preview/index.js

```diff
@@ -165,19 +165,20 @@ const DesignPreview = React.createClass( {
 			return;
 		}
 		event.preventDefault();
-		// TODO: if the href is on the current site, load the href as a preview and fetch markup for that url
 	},
 
 	getPreviewUrl( site ) {
 		if ( ! site ) {
 			return null;
 		}
 
-		const parsed = url.parse( site.options.unmapped_url, true );
+		const slug = this.props.previewSlug ? '/' + this.props.previewSlug : '';
+		const baseUrl = site.options.unmapped_url + slug;
```

Comment:
Alternatively, rather than concatenating at this point, you could assign `parsed.pathname` below to allow `url.format` to concatenate on your behalf.